### PR TITLE
Feat: Filter notes

### PR DESCRIPTION
**Pull request by dikshadhote**
- Sort by Label
- Sort by Priority

image

### CODE REVIEW

File: src/Context/notes-context.js [hunk 1]
```diff
@@ -75,11 +65,17 @@ const NoteProvider = ({ children }) => {
     setNoteData(notes);
   };
 
+  const labelData = noteData.reduce((acc, curr) => {
+    return (acc = [...acc, curr.addLabel]);
+  }, []);
```
Comment (dev-enforced) [inline src/Context/notes-context.js:70]: Can simply write it as 
```js
const labelData=noteData.reduce((acc,curr)=>[...acc,curr.addLabel],[])
```

File: src/Context/notes-context.js [hunk 2]
```diff
@@ -75,11 +65,17 @@ const NoteProvider = ({ children }) => {
...
+
+  const uniqueSet = [...new Set(labelData)];
```
Comment (dev-enforced) [inline src/Context/notes-context.js:72]: Got introduced to this new way of storing unique Labels

**Review by dev-enforced — APPROVED** [review]
Do remove the console logs before pushing to production. Can search on this webpage by doing Ctrl+F and writing console.log in the search inputs part
The functionality is working as expected and the design of the cards also looks awesome with the label and priority chips good work on it @dikshadhote !! Do take care of the refactoring part where you can shift some constant parts of your file be it the initial filter State or priorityList array into a separate constants.js file reducing the number of variables in your context!!

File: src/Context/filter-notes-context.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      }
+      return noteArr;
+    }
+  };
+
+  const filterNotesBySorting = (filterState, ...sortFunctionArr) => {
+    return (noteData) => {
+      return sortFunctionArr.reduce((filteredListAcc, currFilterFunction) => {
+        console.log("acc", filteredListAcc);
+        console.log(currFilterFunction);
```
Comment (tushartiwari7) [inline src/Context/filter-notes-context.js]: remove consoles.

File: src/index.js [hunk 4]
```diff
@@ -16,11 +17,13 @@ ReactDOM.render(
...
-              <App />
-            </TrashProvider>
-          </ArchiveProvider>
+          <FilterProvider>
+            <ArchiveProvider>
+              <TrashProvider>
+                <App />
+              </TrashProvider>
+            </ArchiveProvider>
+          </FilterProvider>
```
Comment (tushartiwari7) [inline src/index.js:26]: Suggestion: Can create a context cluster here.

**Review by tushartiwari7 — APPROVED** [review]
Code looks fine.
liked the compose function and the use of `Set`.
A good pr, you can merge this.

**Review by neetukumari4858 — APPROVED** [review]
code is readable 
everything is well Structured 
good job @dikshadhote

**Pull request merged**